TodoTasker/app/src/androidTest/java/com/backend/todo_tasker/ReminderNotificationShowUpRecordedTest.kt

```diff
@@ -0,0 +1,156 @@
+package com.backend.todo_tasker
+
+
+import android.view.View
+import android.view.ViewGroup
+import android.widget.DatePicker
+import android.widget.TimePicker
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.*
+import androidx.test.espresso.contrib.PickerActions
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.filters.LargeTest
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.ActivityTestRule
+import androidx.test.runner.AndroidJUnit4
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiObject2
+import androidx.test.uiautomator.Until
+import com.backend.todo_tasker.database.DatabaseClass
+import junit.framework.Assert.assertEquals
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers
+import org.hamcrest.Matchers.`is`
+import org.hamcrest.Matchers.allOf
+import org.hamcrest.TypeSafeMatcher
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.*
+
+
+@LargeTest
+@RunWith(AndroidJUnit4::class)
+class ReminderNotificationShowUpRecordedTest {
+
+    @Rule
+    @JvmField
+    var mActivityTestRule = ActivityTestRule(MainActivity::class.java)
+
+    @Test
+    fun reminderNotificationShowUp() {
+        val appContext = InstrumentationRegistry.getInstrumentation().targetContext
+        val notificationTitle =  appContext.getString(R.string.NOTIFICATION_TITLE)
+        val notificationText =  appContext.getString(R.string.NOTIFICATION_TEXT)
+        val timeout:Long =  61 * 1000
```

Comment:
Why exactly this hardcoded value? 61s? Just random pick?

Reply:
Because sometimes if we take < 60 seconds the test will fail, because the notification didn't got delivered. I think this has something to do with this snipped:
![image](https://user-images.githubusercontent.com/57155243/118227969-b5e2f000-b489-11eb-9be4-03c1bb3a8da5.png)
(From BackgroundService line 58 ff.)

Reply:
ah verstehe danke